### PR TITLE
Ncag regex support

### LINKED ISSUE #47: Adding regex support to NCAG
add flags for regex in NCAG model

### PR DESCRIPTION
**Pull request by RA341**
## Proposed changes
--------------------
This resolves #47 and updates the function in #40 to use regex

Adds isRegex fields to model and type, allowing NCAG to use regex.

## Types of changes
--------------------
What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist
--------------------
_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of
them, don't hesitate to ask. This is simply a reminder of what we are going to look for before merging your code._

- [x] My changeset covers only what is described above (no extraneous changes)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments
--------------------

Since I trouble adding is_regex column for nacg in migration files using instructions in the readme

I used  ```npm run typeorm -- schema:log```

Which outputted the query to add is_regex 
``` `ALTER TABLE "nonContainerAutoGrader" ADD "is_regex" boolean NOT NULL;` ```

and I manually added the line in ``devU-api/src/migration/1708215731032-entityFixUp.ts``

```await queryRunner.query(`ALTER TABLE "nonContainerAutoGrader" ADD "is_regex" boolean NOT NULL;`)```

This feels a bit hacky but I tested with populate-db script and everything is working so I think it should be fine.

Also I feel there should be more tests for the grader function but not sure what else to add

### CODE REVIEW

File: devU-api/scripts/populate-db.ts [hunk 1]
```diff
@@ -1,32 +1,33 @@
-const API_URL = "http://localhost:3001"
+const API_URL = 'http://localhost:3001'
 
 //Set to a valid refreshToken, which is set as a cookie after logging in at localhost:9000
-const TOKEN = ""
+const TOKEN =
+  'eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6InNrMDcxMTIwMjEifQ.eyJ1c2VySWQiOjEsImlzUmVmcmVzaFRva2VuIjp0cnVlLCJpYXQiOjE3MDg2NTk1NDYsImV4cCI6MTcwOTUyMzU0NiwiYXVkIjpbImRldlUtYXBpIiwiZGV2VS1jbGllbnQiXSwiaXNzIjoiZGV2VS1hdXRoIiwic3ViIjoiMSJ9.anY1qvejfvqhz73I8Wel7n8LjEh6mJsqnAESJ3miir4jdVuSCA5bwsEjln02l9q1JpwixR3P3J_kbob6yKHGNNk7Nxd3Vw-fYHBQTbpatAKESu2APWJo196GqN1_Rij4uAYzkUt4_TJ_kl1OBEAxx-VjX7-JLk9JWqIsg829NiarRO8qFG-Uyv7br-UaJnmOtFNgaETmYV8TMSY344m7ETPvjJ9VtXkCVl45SBNOp7o3_NGTOAhK6Jlw4cj3L2vsG8x2DH4KEFBsy73hfJv8ONy9y6ypKGEK12votV_G0fhSbGo6-DG9k4Fx1Xucus1By_hQPPg3G43fGWVAWuDv0bEbm68LZBQgio5Bi2L5ndxUa-D5LAFy2PTMFWQeuhsC_pEEhqemQCtU4uBdriE3xiFy0KQmfzQxBOYq8iRPaKoQHmn2hDn3kUyh9Sze92MwdV_T-PAovha4wQee5dtvjmzZUZQVisUjk2CR67EInzSWrcC11BMXWVXvatZjRQ_o\n'
```
Comment (jessehartloff) [inline devU-api/scripts/populate-db.ts:5]: Don't check in your own token

File: devU-api/scripts/populate-db.ts [hunk 2]
```diff
@@ -43,148 +44,278 @@ async function SendPOST(path: string, requestBody: string) {
...
+    //CategoryScores - ROUTE NOT FUNCTIONAL
+
+
+    //CourseScores - ROUTE NOT FUNCTIONAL
+
+    console.log('Script completed successfully!')
+  } catch (e) {
+    console.error(e)
+  }
+
```
Comment (jessehartloff) [inline devU-api/scripts/populate-db.ts:318]: Looks like all linter changers. I guess we've been slacking on the linter

File: devU-api/src/entities/nonContainerAutoGrader/tests/nonContainerAutoGrader.grader.test.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+
+describe('NonContainerAutoGrader grader', () => {
+  let mockQuestionWithoutRegex: NonContainerAutoGrader
+
+  describe('NonContainerAutoGrader without regex', () => {
+    beforeEach(() => {
+      mockQuestionWithoutRegex = {
+        id: 23,
+        question: 'The Answer to the Ultimate Question of Life, the Universe, and Everything',
```
Comment (jessehartloff) [inline devU-api/src/entities/nonContainerAutoGrader/tests/nonContainerAutoGrader.grader.test.ts:13]: Excellent reference

File: devU-api/src/entities/nonContainerAutoGrader/tests/nonContainerAutoGrader.grader.test.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    test('Student correct answer', () => {
+      const studentAnswer = '42'
+      const expectedResult = checkAnswer(studentAnswer, mockQuestionWithoutRegex)
+
+      expect(expectedResult).toEqual(mockQuestionWithoutRegex.score)
+    })
+
+    test('Student messed up', () => {
+      const studentAnswer = '69'
```
Comment (jessehartloff) [inline devU-api/src/entities/nonContainerAutoGrader/tests/nonContainerAutoGrader.grader.test.ts:29]: Not so excellent reference..

File: devU-api/src/migration/1708215731032-entityFixUp.ts [hunk 5]
```diff
@@ -48,6 +48,8 @@ export class entityFixUp1708215731032 implements MigrationInterface {
         await queryRunner.query(`ALTER TABLE "submission_scores" ADD CONSTRAINT "FK_60b6432533ce75c224598af6bf5" FOREIGN KEY ("submission_id") REFERENCES "submissions"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
         await queryRunner.query(`ALTER TABLE "user_courses" ADD CONSTRAINT "FK_7ecb10d15b858768c36d37727f9" FOREIGN KEY ("user_id") REFERENCES "users"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
         await queryRunner.query(`ALTER TABLE "user_courses" ADD CONSTRAINT "FK_d65a2771413a10753d76937b3d6" FOREIGN KEY ("course_id") REFERENCES "courses"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
+        await queryRunner.query(`ALTER TABLE "nonContainerAutoGrader"
+            ADD "is_regex" boolean NOT NULL;`)
```
Comment (jessehartloff) [inline devU-api/src/migration/1708215731032-entityFixUp.ts:52]: Good

**Review by jessehartloff — CHANGES_REQUESTED** [review]
Looks good except for your token being pushed. Remove your token and e're good to merge

File: devU-api/scripts/populate-db.ts (hunk 1 above)
Comment (RA341) [reply devU-api/scripts/populate-db.ts]: fixed and pushed

**Review by jessehartloff — APPROVED** [review]
Good after removing token

**Pull request merged**